### PR TITLE
In-band management via mgmt VRF

### PR DESCRIPTION
**Pull request by venkatmahalingam**
This document covers the design details for in-band management via mgmt VRF,

1) Add in-band management field in MGMT_VRF_CONFIG table
2) Allow "mgmt" VRF name configuration for any L3 interface table
3) Create virtual router in SAI layer for mgmt VRF
4) Create router interface(s) in mgmt VRF context in NPU for mgmt interfaces (e.g mgmt VLAN)

Signed-off-by: Venkatesan Mahalingam <venkatesan_mahalinga@dell.com>

### CODE REVIEW

File: doc/vrf/SONiC_in_band_mgmt_via_mgmt_Vrf_HLD.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+"vrf_name"           = "mgmt" ; Existing field but accepts mgmt VRF name
+
+```
+# Flows
+
+The following diagrams capture the kernel and SAI configuration flows.
+
+## Mgmt VRF configuration flow
+
+![](../../images/vrf_hld/in_band_mgmt_vrf_config_flow.png)
```
Comment (prsunny) [inline doc/vrf/SONiC_in_band_mgmt_via_mgmt_Vrf_HLD.md:47]: I don't see `table_id` part of the VRF Table. Can you please check and remove?

File: doc/vrf/SONiC_in_band_mgmt_via_mgmt_Vrf_HLD.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+## L3 inteface to mgmt VRF bind flow
+
+![](../../images/vrf_hld/in_band_mgmt_vrf_intf_config_flow.png)
```
Comment (prsunny) [inline doc/vrf/SONiC_in_band_mgmt_via_mgmt_Vrf_HLD.md:51]: As we do for loopback, is it only necessary to create IP2ME routes in this case?

Comment (venkatmahalingam) [reply doc/vrf/SONiC_in_band_mgmt_via_mgmt_Vrf_HLD.md:51]: Yes, will program only IP2ME when any interface is part of mgmt VRF like we follow for IP on loopback interface i.e will ignore router-interface creation in the NPU.

Comment (venkatmahalingam) [reply doc/vrf/SONiC_in_band_mgmt_via_mgmt_Vrf_HLD.md:51]: No, we need to create a router interface as well in HW and install IP2ME route

File: doc/vrf/SONiC_in_band_mgmt_via_mgmt_Vrf_HLD.md (hunk 1 above)
Comment (venkatmahalingam) [reply doc/vrf/SONiC_in_band_mgmt_via_mgmt_Vrf_HLD.md:47]: sure, will remove it.

**Review by prsunny — APPROVED** [review]

**Pull request merged**